### DISCUSSION TITLE
Handle case where ID's are server driven

### DISCUSSION THREAD

**Discussion by rrozenvasser-blackbird** [pointfreeco/swift-identified-collections#51] [Q&A]
How do you handle the case where you want to update the UI instantly upon domain model creation but the server is responsible for generating a unique id: 
```
class TodosViewModel: ObservableObject {
  ...
  func createTodo() async {
    let localTodo = Todo(id: UUID().uuidString)
    self.updateOrInsert(todo, at: 0)

    do {
      let serverTodo = try await self.apiClient.createTodo(todo)

      // ❌ Precondition failed: Element identity must remain constant
      self.todos[id: localTodo .id] = serverTodo
    } catch {
      // Handle error
    }
  }
}
```

**Comment by kamcma** [comment]
This gets a little more into service-backed application design writ large and beyond the scope of this library per se. But in the example given, I think you'd want to make the create API call _first_, passing it not an ID-ed `Todo` but some ID-less `TodoParameters` or something like that. Have the API _return_ the ID'ed `Todo`, which you can then insert into the model that drives the UI.
